python/google/protobuf/internal/well_known_types.py

```diff
@@ -244,6 +244,29 @@ def FromDatetime(self, dt):
     self.seconds = calendar.timegm(dt.utctimetuple())
     self.nanos = dt.microsecond * _NANOS_PER_MICROSECOND
 
+  def _comparable_scalar(self, obj):
+    # not dealing with datetime as didn't want to cause any surprises with 2/3
+    # differences. see https://github.com/protocolbuffers/protobuf/pull/5168
+    return obj.ToNanoseconds()
```

Comment:
It comes to a question "do we want to do normalize before compare" . It was discussed in #4318:

Message has implemented an equal compare which compare if all fields are the same, thus there are two equals  "time equality"  and  "protocol buffer equality" for Timestamps. If we do normalize on time equality, the two will not match.  We think it make sense to not normalize for all comparison. Add comments that if the timestamp is not normalized the result may not correct in the API documentation

Reply:
Thanks for the detail @anandolee! Where about would I make changes to get into the API documentation?

_p.s. I should probably update the documentation mention the change of #5234_

Reply:
Add comments in the well_known_types.py should be enough.
Before " def __eq__(self, other):" add something like "# Make sure to normalized timestamps before using the following comparisons. Comparing non-normalized timestamps is not specified and may give unexpected results."

C#  example:
https://github.com/protocolbuffers/protobuf/blob/52fee0e8f03a56931a38e63c2b2ac2ba91191742/csharp/src/Google.Protobuf/WellKnownTypes/TimestampPartial.cs#L229

Reply:
Thanks, I just amended with the comment